qt/widgets/common/inc/MantidQtWidgets/Common/ImageInfoWidget.h

```diff
@@ -0,0 +1,39 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#pragma once
+
+#include "DllOption.h"
+#include "MantidAPI/Workspace_fwd.h"
+#include "MantidQtWidgets/Common/ImageInfoModel.h"
+
+#include <QTableWidget>
+
+namespace MantidQt {
+namespace MantidWidgets {
+
+/**
+ * A table widget containing information about the pixel the mouse is over in
+ * an image
+ */
+class EXPORT_OPT_MANTIDQT_COMMON ImageInfoWidget : public QTableWidget {
+  Q_OBJECT
+
+public:
+  ImageInfoWidget(const Mantid::API::Workspace_sptr &ws,
```

Comment:
I think we are missing a presenter from the MVP triad here? For example the widget is deciding what model should be used and that logic is not tested anywhere. A presenter with the ability to mock view means we can then test more of the presentational logic. It will require an interface for the view but As we only have a single public method then this shouldn't be too much burden. [WorkspacePresenter.cpp](https://github.com/mantidproject/mantid/blob/master/qt/widgets/common/src/WorkspacePresenter/WorkspacePresenter.cpp#L18) is an example of this.